libs/community/langchain_community/retrievers/bm25s.py

```diff
@@ -0,0 +1,137 @@
+from __future__ import annotations
+
+from typing import Any, Dict, Iterable, List, Optional
+
+from langchain_core.callbacks import CallbackManagerForRetrieverRun
+from langchain_core.documents import Document
+from langchain_core.pydantic_v1 import Field
+from langchain_core.retrievers import BaseRetriever
+
+
+class BM25SRetriever(BaseRetriever):
+    """`BM25S` retriever."""
+
+    retriever: Any
+    """ BM25 retriever."""
+    docs: List[Document] = Field(repr=False)
```

Comment:
Why is docs an attribute on the retriever? Why is it needed after the index has been built?

Reply:
I can change it such that it is not required by building the return_docs instead. But this requires having the documents stored in BM25S as a corpus instead. Currently the ID's is returned by BM25S but it can be changed to return the documents instead.